VolumetricConvolution.lua

```diff
@@ -291,119 +78,28 @@ function VolumetricConvolution:createIODescriptors(input)
    end
 end
 
-local one = torch.FloatTensor({1});
-local zero = torch.FloatTensor({0});
-
-local function makeContiguous(self, input, gradOutput)
-   if not input:isContiguous() then
-      self._input = self._input or input.new()
-      self._input:typeAs(input):resizeAs(input):copy(input)
-      input = self._input
-   end
-   if gradOutput and not gradOutput:isContiguous() then
-      self._gradOutput = self._gradOutput or gradOutput.new()
-      self._gradOutput:typeAs(gradOutput):resizeAs(gradOutput):copy(gradOutput)
-      gradOutput = self._gradOutput
-   end
-   return input, gradOutput
-end
-
 function VolumetricConvolution:updateOutput(input)
-   if not self.weightDesc then self:resetWeightDescriptors() end
-   input = makeContiguous(self, input)
-   self:createIODescriptors(input)
-   errcheck('cudnnConvolutionForward', cudnn.getHandle(),
-            one:data(),
-            self.iDesc[0], input:data(),
-            self.weightDesc[0], self.weight:data(),
-            self.convDesc[0], self.fwdAlgType[0],
-            self.extraBuffer:data(), self.extraBufferSizeInBytes,
-            zero:data(),
-            self.oDesc[0], self.output:data());
-   errcheck('cudnnAddTensor', cudnn.getHandle(),
-            one:data(),
-            self.biasDesc[0], self.bias:data(), one:data(),
-            self.oDescBias[0], self.output:data());
-   return self.output
+   return Convolution:updateOutput(input)
```

Comment:
There are multiple instances where `Convolution:funcname(...)` is used, which I believe should be `Convolution.funcname(self, ...)`